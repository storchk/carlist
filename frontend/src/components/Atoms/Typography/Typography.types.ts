import type { LabelHTMLAttributes } from 'react'

import type { FontSize, FontWeights } from '@/styles'
import type { ColorNames } from '@/theme'

export type TypographyProps = {
  color?: ColorNames | 'inherit'
  fontWeight?: FontWeights
  fontSize?: {
    mobile?: FontSize
    tablet?: FontSize
    desktop?: FontSize
  }
  tag?: keyof JSX.IntrinsicElements
} & LabelHTMLAttributes<HTMLElement>
