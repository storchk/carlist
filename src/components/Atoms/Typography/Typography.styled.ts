import styled, { css } from 'styled-components'

import { mediaQuery } from '../../../styles'
import type { TypographyProps } from './Typography.types'

export const StyledTypography = styled.div<TypographyProps>(
  ({ theme, fontWeight = 'regular', fontSize, color }) => css`
    font-weight: ${theme.font.weight[fontWeight]};
    color: ${color && color !== 'inherit' ? theme.colors[color] : 'inherit'};

    font-size: ${fontSize?.mobile ? theme.font.size[fontSize.mobile] : 'inherit'};
    ${fontSize?.tablet
      ? css`
          ${mediaQuery.md} {
            font-size: ${theme.font.size[fontSize.tablet]};
          }
        `
      : ''}

    ${fontSize?.desktop
      ? css`
          ${mediaQuery.lg} {
            font-size: ${theme.font.size[fontSize.desktop]};
          }
        `
      : ''}
  `
)
