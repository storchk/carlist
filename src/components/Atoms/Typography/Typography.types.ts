import type { LabelHTMLAttributes } from 'react'

import type { FontSize, FontWeights } from '../../../styles/fonts'
import type { ColorNames } from '../../../theme/colors'

export type TypographyProps = {
  color?: ColorNames | 'inherit'
  fontWeight?: FontWeights
  fontSize?: {
    mobile?: FontSize
    tablet?: FontSize
    desktop?: FontSize
  }
  tag?: keyof JSX.IntrinsicElements
} & LabelHTMLAttributes<HTMLElement>
