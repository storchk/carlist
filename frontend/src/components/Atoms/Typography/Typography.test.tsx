import { axe } from 'jest-axe'

import { testing } from '../../../testing'
import { Typography } from './Typography'
import type { TypographyProps } from './Typography.types'

const { render, renderer } = testing
const mockText = 'Lorem Ipsum'

const renderComponent = (props?: Partial<TypographyProps>) => render(<Typography {...props} />)

describe('atoms: Typography', () => {
  it('snapshot', () => {
    const view = renderer(<Typography>{mockText}</Typography>)
    expect(view).toMatchSnapshot()
  })
  it('a11y - No violations detected', async () => {
    const { container } = renderComponent()
    await expect(axe(container)).resolves.toHaveNoViolations()
  })
})
