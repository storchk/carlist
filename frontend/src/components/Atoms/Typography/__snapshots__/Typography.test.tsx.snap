// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms: Typography snapshot 1`] = `
.c0 {
  font-weight: 400;
  color: inherit;
  font-size: inherit;
}

<span
  className="c0"
>
  Lorem Ipsum
</span>
`;
